gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/JmsDestinationLinker.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2018. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayexport.tasks.explode.linker;
+
+import com.ca.apim.gateway.cagatewayconfig.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.beans.JmsDestination;
+import com.ca.apim.gateway.cagatewayconfig.beans.StoredPassword;
+import org.jetbrains.annotations.NotNull;
+
+import javax.inject.Singleton;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.VariableUtils.extractVariableName;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Singleton
+public class JmsDestinationLinker implements EntityLinker<JmsDestination> {
+
+    private static final Pattern secpassPattern = Pattern.compile("secpass.(.+?).plaintext");
+    
+    @Override
+    public Class<JmsDestination> getEntityClass() {
+        return JmsDestination.class;
+    }
+
+    @Override
+    public void link(JmsDestination entity, Bundle bundle, Bundle targetBundle) {
+        linkJndiStoredPassword(entity, bundle);
+        linkDestinationStoredPassword(entity, bundle);
+        
+        // TODO: link private key(s), service
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWdWyJqwpqfxNGPa-4iS&open=AWdWyJqwpqfxNGPa-4iS)

[ISSUE_KEY]: AWdWyJqwpqfxNGPa-4iS